### PR TITLE
Update polygon request

### PR DESCRIPTION
**Pull request by ManuelStardust**
Update the request for the query of loss and persistence by polygon according to the current script.

### CODE REVIEW

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts [hunk 1]
```diff
@@ -88,10 +89,12 @@ export class ForestLossPersistenceController {
           const forestLPArea = rawData
             .filter((d: { period: string }) => d.period === periods[0])
             .reduce((totalArea, periodData) => totalArea + periodData.area, 0);
+          const layersFolder = data.files.dir_png;
```
Comment (cazapatamar) [inline src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts:92]: Consider handle the case when `data.files` is null or undefined

**Review by cazapatamar — CHANGES_REQUESTED** [review]

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts (hunk 1 above)
Comment (ManuelStardust) [reply src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts]: Done

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts [hunk 2]
```diff
@@ -88,10 +89,12 @@ export class ForestLossPersistenceController {
           const forestLPArea = rawData
             .filter((d: { period: string }) => d.period === periods[0])
             .reduce((totalArea, periodData) => totalArea + periodData.area, 0);
+          const layersFolder = data.files.dir_png ?? "";
```
Comment (cazapatamar) [inline src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts:92]: This solution handle the case when `dir_png` is null or undefined but not when `files` is null or undefined. Consider add the operator `?` (optional chaining operator) on `files` property.

**Review by cazapatamar — CHANGES_REQUESTED** [review]

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts (hunk 2 above)
Comment (ManuelStardust) [reply src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts]: Done

File: src/pages/search/SearchContext.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ export interface SearchContextValues {
   geofenceId: string | number;
   searchType: "definedArea" | "drawPolygon";
   polygon: Polygon | null;
-  setPolygonArea(value: number): void;
+  setPolygonValues(areaValue: number, folder?: string): void;
```
Comment (erikasv) [inline src/pages/search/SearchContext.tsx:9]: `folder` shouldn't be optional, you are already making it an string always in the FLP controller

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ interface ForestLPData {
   forestLP: Array<ForestLPExt>;
   forestPersistenceValue: number;
   forestLPArea?: number;
+  layersFolder?: string;
```
Comment (erikasv) [inline src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts:17]: Same comment as the type in the search context

**Review by erikasv — CHANGES_REQUESTED** [review]

File: src/pages/search/SearchContext.tsx (hunk 3 above)
Comment (ManuelStardust) [reply src/pages/search/SearchContext.tsx]: Done

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts (hunk 4 above)
Comment (ManuelStardust) [reply src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts:17]: In this case ForestLPData is used as type for response the response of biotablero's backend  and boninabox's backend, however biotablero's backend doesn't return a value for the folder.

File: src/pages/search/SearchContext.tsx (hunk 3 above)
Comment (ManuelStardust) [reply src/pages/search/SearchContext.tsx]: Done

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts (hunk 4 above)
Comment (erikasv) [reply src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts:17]: you're right

**Review by erikasv — APPROVED** [review]

**Review by cazapatamar — APPROVED** [review]

**Pull request merged**